Assets/Scripts/Managers/AudioManager.cs

```diff
@@ -0,0 +1,78 @@
+using UnityEngine.Audio;
+using System;
+using UnityEngine;
+
+
+public class AudioManager : MonoBehaviour
+{
+    public static AudioManager Instance;
+    [SerializeField] public AudioMixerGroup musicMixerGroup;
+    [SerializeField] public AudioMixerGroup sfxMixerGroup;
+    [SerializeField] public Sound[] sounds;
+
+    void Awake()
+    {
+
+        foreach (Sound s in sounds)
+        {
+            s.source = gameObject.AddComponent<AudioSource>();
+            s.source.clip = s.clip;
+            
+            s.source.volume = s.volume;
+            s.source.pitch = s.pitch;
+            s.source.loop = s.loop;
+            s.source.playOnAwake = s.playOnAwake;
```

Comment:
As previously mentioned - please put in it's own function in `Sound.cs`

Reply:
Clarification:
This would be better served as 
```cs
foreach (Sound s in sounds)
{
    s.Setup();
}
```
with a `Setup()` function as follows in `Sound.cs`
```cs
void Setup()
{
    source = gameObject.AddComponent<AudioSource>();
    source.clip = clip;
    source.volume = volume;
    source.pitch = pitch;
    source.loop = loop;
    source.playOnAwake = playOnAwake
}
```